pkg/controller/operators/catalog/operator.go

```diff
@@ -429,17 +413,22 @@ func (o *Operator) syncCatalogSources(obj interface{}) (syncError error) {
 		logger.Debug("catsrc configmap state good, checking registry pod")
 	}
 
-	reconciler := o.reconciler.ReconcilerForSource(catsrc)
-	if reconciler == nil {
+	srcReconciler := o.reconciler.ReconcilerForSource(catsrc)
+	if srcReconciler == nil {
 		// TODO: Add failure status on catalogsource and remove from sources
 		return fmt.Errorf("no reconciler for source type %s", catsrc.Spec.SourceType)
 	}
 
+	healthy, err := srcReconciler.CheckRegistryServer(catsrc)
```

Comment:
this is the bulk of the feature change :) everything else is fixing flakes in the tests.